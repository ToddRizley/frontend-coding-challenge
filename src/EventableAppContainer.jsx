import React, { Component } from 'react';
import 'whatwg-fetch';
import EventListContainer from './EventListContainer';
import AddEventForm from './AddEventForm';
import SearchForm from './SearchForm';
import EventSorter from './EventSorter'
import {validateDates, sortList} from './utils.js';

class EventableAppContainer extends Component {

  constructor(props){
    super(props)
    this.state = {
      eventList: {},
      currentListView: {}
    }
  };

  searchByTitle(event){
    event.preventDefault()
    var filterBy = String(event.target.value).toLowerCase()
    var length = filterBy.length

    if ( length > 0 ) {
      var filterEvents = this.state.currentListView.filter( (event)=> {
        return (event.title.slice(0, length).toLowerCase() === filterBy)
        })
      this.setState({currentListView: filterEvents})
    } else {
      this.setState({currentListView: this.state.eventList})
    }
  };


  handleSubmit(event){
    event.preventDefault()
    let title = event.target[0].value
    let startTime = event.target[1].value
    let endTime = event.target[2].value
    if(validateDates(startTime, endTime)) {
      var eventList = this.state.eventList
      eventList.push({title: title, start_time: startTime, end_time: endTime})
      this.setState({eventList: eventList})
    }
  };

  handleChange(event){
    var sortType = event.target.value
    var sortedList = sortList(sortType, this.state.currentListView, this.state.eventList)
    this.setState({currentListView: sortedList})
  };

  componentWillMount(){
    const APITOKEN = "Token 7761e7e3b25a1d6d315901fcd7180d971f77ea2e"
    const URL = 'https://api.eventable.com/v1/events/'
    var obj = {
      method: 'GET',
      headers: {
        'Authorization': APITOKEN
      }
    };

  fetch(URL, obj)
  .then( (res)=> {
    return res.json();
   })
  .then( (resJson)=> {
    this.setState({eventList: resJson['results'], currentListView: resJson['results']});
  });
 };

  render() {
    return (
      <div className='col-xs-12'>
        <AddEventForm
          handleSubmit={this.handleSubmit.bind(this)}
        />
        <SearchForm
          searchByTitle={this.searchByTitle.bind(this)}
        />
        <EventSorter
          handleChange={this.handleChange.bind(this)}
        />
        <EventListContainer
          currentListView={this.state.currentListView}
        />
      </div>
    );
  };
};

export default EventableAppContainer;
