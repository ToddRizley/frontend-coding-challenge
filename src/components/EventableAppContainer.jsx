import React, { Component } from 'react';
import 'whatwg-fetch';
import EventListContainer from './EventListContainer';
import AddEventForm from './AddEventForm';
import SearchForm from './SearchForm';
import EventSorter from './EventSorter';
import { validateDates, sortList } from '../utils';

class EventableAppContainer extends Component {

  constructor(props) {
    super(props);
    this.state = {
      eventList: {},
      currentListView: {},
    };
  }

  componentWillMount() {
    const APITOKEN = 'Token 7761e7e3b25a1d6d315901fcd7180d971f77ea2e';
    const URL = 'https://api.eventable.com/v1/events/';
    const obj = {
      method: 'GET',
      headers: {
        Authorization: APITOKEN,
      },
    };

    fetch(URL, obj)
  .then(res => res.json())
  .then((resJson) => {
    this.setState({ eventList: resJson.results, currentListView: resJson.results });
  });
  }

  searchByTitle(event) {
    event.preventDefault();
    const filterBy = String(event.target.value).toLowerCase();
    const length = filterBy.length;

    if (length > 0) {
      const filterEvents =
      this.state.currentListView.filter(
        eventObj => (eventObj.title.slice(0, length).toLowerCase() === filterBy));
      this.setState({ currentListView: filterEvents });
    } else {
      this.setState({ currentListView: this.state.eventList });
    }
  }


  handleSubmit(event) {
    event.preventDefault();
    const title = event.target[0].value;
    const startTime = event.target[1].value;
    const endTime = event.target[2].value;
    if (validateDates(startTime, endTime)) {
      const eventList = this.state.eventList;
      eventList.push({ title, start_time: startTime, end_time: endTime });
      this.setState({ eventList });
    }
  }

  handleChange(event) {
    const sortType = event.target.value;
    const sortedList = sortList(sortType, this.state.currentListView, this.state.eventList);
    this.setState({ currentListView: sortedList });
  }

  render() {
    return (
      <div className="col-xs-12">
        <SearchForm
          searchByTitle={this.searchByTitle.bind(this)}
        />
        <EventSorter
          handleChange={this.handleChange.bind(this)}
        />
        <EventListContainer
          currentListView={this.state.currentListView}
        />
        <AddEventForm
          handleSubmit={this.handleSubmit.bind(this)}
        />
      </div>
    );
  }
}

export default EventableAppContainer;
