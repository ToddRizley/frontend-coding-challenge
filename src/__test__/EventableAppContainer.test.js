import React from 'react';
import EventableAppContainer from '../EventableAppContainer'
import renderer from 'react-test-renderer';

describe('<EventableAppContainer/>', () => {
   it('renders correctly', () => {
    const rendered = renderer.create(
      <EventableAppContainer />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
