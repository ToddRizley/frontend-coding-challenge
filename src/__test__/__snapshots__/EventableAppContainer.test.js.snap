// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventableAppContainer/> renders correctly 1`] = `
<div
  className="col-xs-12"
>
  <div
    className="event-form row"
  >
    <h4>
      Add New Event
    </h4>
    <form
      className="form-group row col-xs-12"
      onSubmit={[Function]}
    >
      <div
        className="col-xs-3"
      >
        Event Name: 
        <input
          className="form-control"
          name="event-title"
          placeholder="Add Event Title"
          required={true}
          type="text"
        />
         
      </div>
      <div
        className="col-xs-3"
      >
        Start: 
        <input
          className="form-control"
          name="event-start"
          placeholder="Start Date/Time"
          required={true}
          type="datetime-local"
        />
         
      </div>
      <div
        className="col-xs-3"
      >
        End: 
        <input
          className="form-control"
          name="event-end"
          placeholder="End Date/Time"
          required={true}
          type="datetime-local"
        />
          
      </div>
      <div
        className="col-xs-3"
      >
        <input
          id="add-event-button"
          type="submit"
          value="Submit"
        />
      </div>
    </form>
  </div>
  <form
    className="search-form col-xs-6"
  >
    <div
      className="form-group has-feedback"
    >
      <label
        className="control-label"
      >
        Search Events By Title
      </label>
      <input
        className="form-control"
        onChange={[Function]}
        placeholder="Search by event title"
        type="text"
      />
      <i
        className="glyphicon glyphicon-search form-control-feedback"
      />
    </div>
  </form>
  <div
    className="event-sorter col-xs-6"
  >
    <label
      className="sorter-label"
    >
      Search Events By Title
    </label>
    <select
      className="col-xs-6"
      onChange={[Function]}
    >
      <option
        value="startTime"
      >
        Sort By Start Time
      </option>
      <option
        value="title"
      >
        Sort By Name
      </option>
    </select>
  </div>
  <div
    className="event-list col-xs-12"
  >
    <div
      className="event-list col-xs-12"
    >
      <h3
        className="col-xs-6"
      >
        Event List
      </h3>
    </div>
  </div>
</div>
`;
